tools/ddsrouter_tool/test/compose/validate_subscriber.py

```diff
@@ -0,0 +1,209 @@
+    parser.add_argument(
+        '--allow-duplicates',
+        action='store_true',
+        help='Allow receive duplicated data.'
+    )
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        help='Print test debugging info.'
+    )
+
+    return parser.parse_args()
+
+
+def run(command, timeout):
+    """
+    Run command with timeout.
+
+    :param command: Command to run in list format
+    :param timeout: Timeout for the process
+    :return:
+        - ret_code - The process exit code
+        - stdout - Output of the process
+        - stderr - Error output of the process
+    """
+    ret_code = 0
+
+    proc = subprocess.Popen(command,
+                            stdout=subprocess.PIPE,
+                            universal_newlines=True)
+    try:
+        proc.wait(timeout=timeout)
+    except subprocess.TimeoutExpired:
+        logger.info('Timeout expired. '
+                    'Killing subscriber before receiving all samples...')
+        ret_code = 1
+        proc.send_signal(signal.SIGINT)
+
+    stdout, stderr = proc.communicate()
+
+    return (ret_code, stdout, stderr)
+
+
+def parse_output(data):
+    """
+    Transform the output of the program in a list of received messages.
+
+    :param data: Process stdout
+    :return: List of received messages
+    """
+    regex = re.compile('^Message\sHelloWorld\s+\d+\sRECEIVED$')
+    lines = data.splitlines()
+    filtered_data = [line for line in lines if regex.match(line)]
+
+    return filtered_data
+
+
+def find_duplicates(data):
+    """
+    Find duplicates in a list os strings.
+
+    :param data: List of strings
+    :return: List of tuples with the index of the duplicated strings
+    """
+    duplicates = []
+    lines_seen = {}
+
+    for idx, line in enumerate(data):
+        if line not in lines_seen:
+            lines_seen[line] = idx
+        else:
+            duplicates.append((lines_seen[line], idx))
+
+    if duplicates:
+        logger.info('Found duplicated messages')
+    else:
+        logger.debug('None duplicated messages found')
+
+    return duplicates
+
+
+def validate(command, samples, timeout, allow_duplicates=False):
+    """
+    Validate the output of a subscriber run.
+
+    :param command: Command to run in list format
+    :param samples: Number of samples to receive
+    :param timeout: Timeout for the process
+    :param allow_duplicates: Allow duplicated messages (Default: False)
+    :return: The exit code
+    """
+    ret_code, stdout, stderr = run(command, timeout)
+
+    if ret_code:
+        logger.error('Subscriber application exited with '
+                     f'return code {ret_code}')
+
+        logger.error(f'Subscriber output: \n {stdout}')
+        return ret_code
+    else:
+        data_received = parse_output(stdout)
+
+        logger.debug(f'Subscriber received... \n {data_received}')
+
+        if not allow_duplicates:
+            return len(find_duplicates(data_received))
```

Comment:
Do not return `len` as it will be the exit of the executable, and would be difficult to interpret

Reply:
To discuss...